pkg/devfile/adapters/common/types.go

```diff
@@ -31,6 +33,8 @@ type PushParameters struct {
 	Debug                    bool                    // Runs the component in debug mode
 	DebugPort                int                     // Port used for remote debugging
 	RunModeChanged           bool                    // It determines if run mode is changed from run to debug or vice versa
+	RandomPorts              bool                    // True to forward containers ports on local random ports
```

Comment:
I guess both `RandomPorts` and `ErrOut` need to be declared in `watch.WatchParameters` as well, and handled in the `watch.processEvents` function when recreating a new `PushParameters` object.

Otherwise, when running `odo dev --random-ports` and modifying the container Devfile component during this session, the local port is regenerated, but no longer randomly (I was expecting a new random port, but I got 40001)..

```
❯ odo dev --random-ports
...
↪ Deploying to the cluster in developer mode
 ✓  Waiting for Kubernetes resources [52s]
 ✓  Syncing files into the container [96ms]
 ✓  Building your application in container on cluster (command: build) [458ms]
 •  Executing the application (command: run-with-sleep)  ...
 - Forwarding from 127.0.0.1:43797 -> 8080

Your application is now running on the cluster

Watching for changes in the current directory /home/asoro/work/tmp/go
Press Ctrl+c to exit `odo dev` and delete resources from the cluster

# Changing container component memory limit in Devfile

File /home/asoro/work/tmp/go/devfile.yaml changed
Pushing files...

 ✗  Executing the application (command: run-with-sleep) [2m]
 ✓  Waiting for Kubernetes resources [33s]
 ✓  Syncing files into the container [95ms]
 ✓  Building your application in container on cluster (command: build) [179ms]
 •  Executing the application (command: run-with-sleep)  ...
 - Forwarding from 127.0.0.1:40001 -> 8080

Watching for changes in the current directory /home/asoro/work/tmp/go
Press Ctrl+c to exit `odo dev` and delete resources from the cluster
```

Reply:
Yes, good catch, thanks